docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -0,0 +1,129 @@
+include::./attributes.adoc[]
+= {project-name} - Tips for writing native applications
+
+This guide contains various tips and tricks for getting around various problems that might arise when attempting to run java apps as native images.
+
+== Register reflection
+
+Due to the fact that when building a native image GraalVM operates with a closed world assumption (which enables it to remove all unused code, which in turns yields multiple benefits), reflection targets need to explicitly declared.
+
+An example of failing to explicitly specify reflection is the following:
+
+[source]
+----
+Caused by: org.xml.sax.SAXException: SAX2 driver class com.sun.org.apache.xerces.internal.parsers.SAXParser not found
+java.lang.ClassNotFoundException: com.sun.org.apache.xerces.internal.parsers.SAXParser
+at org.xml.sax.helpers.XMLReaderFactory.loadClass(XMLReaderFactory.java:230)
+----
+
+If one were to execute the native image generation process manually, the `-H:ReflectionConfigurationFiles=` flag would be used to point to a json configuration file that specifies the program elements that would be accessed reflectively.
+
+Quarkus however makes registration of reflection a breeze by using the `ReflectiveClassBuildItem`, thus eliminating the need for such a json configuration file.
+
+To solve the problem shown in the example above one would simply have to declare:
+
+[source,java]
+----
+    @BuildStep
+    ReflectiveClassBuildItem reflection() {
```

Comment:
I think you should say that people need to create a class containing this `@BuildStep`. Otherwise, we don't know what to do with it.